Src/ILGPU.Algorithms/XMath/Cbrt.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            // Perform "Safety Check" for; -n, 0, +Inf, -Inf, NaN
+            if (checksafety)
+            {
+                if (n < 0) { return -Cbrt(-n); }
+                if (n == 0 || double.IsNaN(n) || double.IsInfinity(n)) { return n; }
+            }
+
+            // Initial approximation
+            // Convert the binary representation of float into a positive int
+            // Isolate & Convert mantissa into actual power it represents
+            // Perform cube root on 2^P, to appoximate x using power law
+            double x = 1 << (int)((((*(uint*)&n) >> 53) - 1023) * 0.33333333333333333333f) + 0b1;
+
+            // Perform check if x^3 matches n
+            double xcubed = x * x * x;
+            if (xcubed == n) { return x; }
+
+            // Perform 4 itterations of Halley algorithm for double accuracy
+            double xcubedPlusN = xcubed + n;
+            x = x * ((xcubedPlusN + n) / (xcubed + xcubedPlusN));
+
+            xcubed = x * x * x;
+            xcubedPlusN = xcubed + n;
+            x = x * ((xcubedPlusN + n) / (xcubed + xcubedPlusN));
+
+            xcubed = x * x * x;
+            xcubedPlusN = xcubed + n;
+            x = x * ((xcubedPlusN + n) / (xcubed + xcubedPlusN));
+
+            xcubed = x * x * x;
+            xcubedPlusN = xcubed + n;
+            x = x * ((xcubedPlusN + n) / (xcubed + xcubedPlusN));
+
+            xcubed = x * x * x;
+            xcubedPlusN = xcubed + n;
+            x = x * ((xcubedPlusN + n) / (xcubed + xcubedPlusN));
+
+            xcubed = x * x * x;
+            xcubedPlusN = xcubed + n;
+            x = x * ((xcubedPlusN + n) / (xcubed + xcubedPlusN));
+
+            return x;
+        }
+
+        /// <summary>
+        /// Computes 3rd root of n to within single precision.
+        /// </summary>
+        /// <param name="n">This value's cube root will be calculated.</param>
+        /// <param name="checksafety">Enables safety checks.
+        /// <para>Default : Enabled / true</para>
+        /// <para>This includes checking n for negative and 0 values, Infinities and NaN.</para>
+        /// <para>Set to false at own risk for minor speed improvement to disable safety checks.</para>
+        /// </param>
+        /// <returns>Cbrt(n).</returns>
+        public unsafe static float CbrtFast(float n, bool checksafety = true)
+        {
+            if (checksafety)
+            {
+                if (n < 0) { return -CbrtFast(-n); }
+                if (n == 0 || double.IsNaN(n) || double.IsInfinity(n)) { return n; }
+            }
+
+            // Initial approximation
+            // Convert the binary representation of float into a positive int
+            // Isolate & Convert mantissa into actual power it represents
+            // Perform cube root on 2^P, to appoximate x using power law
+            float x = 1 << (int)((((*(uint*)&n) >> 23) - 127) * 0.33333333333333333333f) + 0b1;
```

Comment:
nit: same here

Reply:
Same as the original comment. I will test all overloads with the interop stuff instead of unsafe code.